### DISCUSSION TITLE
Saving Video Inference Results and Object Tracking

### DISCUSSION THREAD

**Discussion by AliButtarPG** [obss/sahi#641] [Q&A] [answered]
Hi,

1. Video Inference runs great and the results when visualized in a video are great too, but how do I access all those inferences in JSON or a pickle file. I did `export_pickle = True` but the generated pickle file doesn't have all the instances. Is it only for the last frame of the video?

3. My main problem is counting hundreds of small objects shot in a video, is there any tracking functionality with SAHI or can you point me towards a direction where I can use SAHI in this way?

**Comment by fcakyon** [comment]
You found a bug in pickle export when input is a video! This PR should fix the issue: https://github.com/obss/sahi/pull/481. 

Can you please uninstall SAHI:
```pip uninstall sahi```

Install SAHI from branch:
```pip install git+htpps://github.com/obss/sahi@fix-pickle-export-for-video```

Then try again and check if frame-based pickle export is working?

**Comment by fcakyon** [comment]
Currently, SAHI does not include any built-in tracking function, but you should be able to integrate any tracker easily.

**Comment by AliButtarPG** [comment]
Hi @fcakyon,

Thanks for the fix, I have different pickle files for each of the frames now.

Secondly, when you say I should be able to integrate easily, do you have something in mind? I am having difficulty thinking of a process on how to make it all work.

**Comment by fcakyon** [comment, marked as answer]
The easiest way of implementing a tracker on top of SAHI is Norfair. Check this Small object tracking demo: https://github.com/tryolabs/norfair/tree/master/demos/sahi

**Comment by AliButtarPG** [comment]
Thank you for the lead. I will look into it.
